src/auth/guards/roles.guard.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { Reflector } from '@nestjs/core';
+import { JwtService } from '@nestjs/jwt';
+import { Request } from 'express';
+import { CustomerRole } from 'src/customers/domain/customer-role.entity';
+import { Roles } from '../decorators/roles.decorator';
+
+@Injectable()
+export class RolesGuard implements CanActivate {
```

Comment:
**suggestion:** note you can just have an auth guard, later on you can use a decorator for the Roles which takes the current request context and checks if the user provided in that request has those roles.

Moreover, you can set that auth guard to be global, and create a Public decorator only for the register endpoint so you don't keep adding to all request the UseGuards decorator

Reply:
![image](https://github.com/user-attachments/assets/141059e8-8a72-4f17-b92f-f96383e5a30b)

Divided the Validation in two different guards: 

    1. one for effectively checking the JWT and binding the payload to the request
    2. one for checking the role of the user (if role restriction is specified in the @Roles decorator)

Reply:
Also, using the @Public decorator to cancel the global effect of both JwtGuard and RolesGuard.